package/tests/test_api/test_graphql.py

```diff
@@ -1,5 +1,184 @@
+from kedro.io import DataCatalog, Version
+from strawberry import ID
+
+from kedro_viz.api.graphql import JSONObject, TrackingDataSet, get_run_tracking_data
+from kedro_viz.data_access.managers import DataAccessManager
+
+
+@pytest.fixture
+def save_version():
+    yield "2021-11-02T18.24.24.379Z"
+
+
+@pytest.fixture
+def example_tracking_catalog(save_version):
+    # Note - filepath is assigned without using tmp_path as it fails on windows build.
+    # This is a temp soln and will be cleaned up in the future.
+    metrics_dataset = MetricsDataSet(
+        filepath="test.json",
+        version=Version(None, save_version),
+    )
+    metrics_dataset.save({"col1": 1, "col2": 2, "col3": 3})
+
+    dataset = CSVDataSet(filepath="dataset.csv")
+
+    more_metrics = MetricsDataSet(
+        filepath="metrics.json",
+        version=Version(None, save_version),
+    )
+    more_metrics.save({"col4": 4, "col5": 5, "col6": 6})
+
+    json_dataset = JSONDataSet(
+        filepath="tracking.json",
+        version=Version(None, save_version),
+    )
+    json_dataset.save({"col7": "column_seven", "col2": True, "col3": 3})
+
+    catalog = DataCatalog(
+        data_sets={
+            "metrics": metrics_dataset,
+            "csv": dataset,
+            "more_metrics": more_metrics,
+            "json_tracking": json_dataset,
+        }
+    )
+
+    yield catalog
+
+
+@pytest.fixture
+def example_tracking_output(save_version):
+    yield [
+        TrackingDataSet(
+            datasetName="metrics",
+            datasetType="kedro.extras.datasets.tracking.metrics_dataset.MetricsDataSet",
+            data={
+                "col1": [{"runId": save_version, "value": 1.0}],
+                "col2": [{"runId": save_version, "value": 2.0}],
+                "col3": [{"runId": save_version, "value": 3.0}],
+            },
+        ),
+        TrackingDataSet(
+            datasetName="more_metrics",
+            datasetType="kedro.extras.datasets.tracking.metrics_dataset.MetricsDataSet",
+            data={
+                "col4": [{"runId": save_version, "value": 4.0}],
+                "col5": [{"runId": save_version, "value": 5.0}],
+                "col6": [{"runId": save_version, "value": 6.0}],
+            },
+        ),
+        TrackingDataSet(
+            datasetName="json_tracking",
+            datasetType="kedro.extras.datasets.tracking.json_dataset.JSONDataSet",
+            data={
+                "col2": [{"runId": save_version, "value": True}],
+                "col3": [{"runId": save_version, "value": 3}],
+                "col7": [{"runId": save_version, "value": "column_seven"}],
+            },
+        ),
+    ]
+
+
+@pytest.fixture(scope="class", autouse=True)
+def cleanup(request):
+    """Cleanup a testing directory once we are finished."""
+
+    def remove_test_dir():
+        shutil.rmtree("test.json")
+        shutil.rmtree("metrics.json")
+        shutil.rmtree("tracking.json")
```

Comment:
Nit: I'd suggest adding `ignore_errors=True` in case they don't exist for whatever reason.

Also you don't need to have a dedicated cleanup fixture. You can do all of this after `yield catalog` in `example_tracking_catalog` and it would be called after the test. This way is fine too of course.